kie-server-parent/kie-server-controller/kie-server-controller-impl/src/main/java/org/kie/server/controller/impl/KieServerInstanceManager.java

```diff
@@ -55,189 +54,260 @@ public static KieServerInstanceManager getInstance() {
+                                         final ContainerSpec containerSpec) {
+        final RemoteKieServerOperation<Void> operation = getContainersRemoteOperation(serverTemplate,
+                                                                                      containerSpec);
 
-                ServiceResponse<ReleaseId> response = client.updateReleaseId(containerSpec.getId(), containerSpec.getReleasedId());
-                if (!response.getType().equals(ServiceResponse.ResponseType.SUCCESS)) {
-                    logger.debug("Container {} failed to upgrade on server instance {} due to {}", containerSpec.getId(), container.getUrl(), response.getMsg());
-                }
-                collectContainerInfo(containerSpec, client, container);
-                return null;
-            }
-        });
+        return callRemoteKieServerOperation(serverTemplate,
+                                            containerSpec,
+                                            operation);
     }
 
-    public List<Container> getContainers(final ServerTemplate serverTemplate, final ContainerSpec containerSpec) {
-
-        return callRemoteKieServerOperation(serverTemplate, containerSpec, new RemoteKieServerOperation<Void>(){
+    RemoteKieServerOperation<Void> getContainersRemoteOperation(final ServerTemplate serverTemplate,
+                                                                final ContainerSpec containerSpec) {
+        return new RemoteKieServerOperation<Void>() {
             @Override
-            public Void doOperation(KieServicesClient client, Container container) {
-
-                if (containerSpec.getStatus().equals(KieContainerStatus.STARTED)) {
-                    ServiceResponse<KieContainerResource> response = client.getContainerInfo(containerSpec.getId());
-                    if (response.getType().equals(ServiceResponse.ResponseType.SUCCESS)) {
-                        KieContainerResource containerResource = response.getResult();
+            public Void doOperation(KieServicesClient client,
+                                    Container container) {
 
-                        container.setContainerSpecId(containerResource.getContainerId());
-                        container.setContainerName(containerResource.getContainerId());
-                        container.setResolvedReleasedId(containerResource.getResolvedReleaseId() == null ? containerResource.getReleaseId() : containerResource.getResolvedReleaseId());
-                        container.setServerTemplateId(serverTemplate.getId());
-                        container.setStatus(containerResource.getStatus());
-                        container.setMessages(containerResource.getMessages());
+                final ServiceResponse<KieContainerResource> response = client.getContainerInfo(containerSpec.getId());
+                final KieContainerResource containerResource = response.getResult();
```

Comment:
essentially we simply return data regardless of the container status, did I got it right?